scm/github.go

```diff
@@ -111,10 +111,40 @@ func (s *GithubSCM) CreateRepository(ctx context.Context, opt *CreateRepositoryO
 	}, nil
 }
 
+// GetRepository implements the SCM interface.
+func (s *GithubSCM) GetRepository(ctx context.Context, opt *GetRepoOptions) (*Repository, error) {
+	if !opt.valid() {
+		s.logger.Errorf("GetRepository got invalid GetRepoOptions: %+v", opt)
+		return nil, ErrMissingFields
+	}
+	var gitRepo *github.Repository
+	var err error
+	// if ID is set, get by ID
+	if opt.ID > 0 {
+		gitRepo, _, err = s.client.Repositories.GetByID(ctx, opt.ID)
+	} else {
+		// otherwise get by repo name and owner (usually owner = organization name)
+		gitRepo, _, err = s.client.Repositories.Get(ctx, opt.Owner, opt.Path)
+	}
+	if err != nil {
+		return nil, err
```

Comment:
Shouldn't we log err here?

Reply:
I tried to move most of the logging to API level to avoid spamming with the same error from several packages, only left cases when this is necessary. I'll recheck this one.